front/pages/api/w/[wId]/dust_app_secrets/index.ts

```diff
@@ -82,49 +83,38 @@ async function handler(
           return {
             createdAt: s.createdAt.getTime(),
             name: s.name,
-            value: clearSecret,
+            value: redactString(clearSecret, 1),
           };
         }),
       });
       return;
 
-    case "DELETE":
-      const { name: deleteSecretName } = req.body;
-      const secret = await DustAppSecret.findOne({
-        where: {
-          name: deleteSecretName,
-          workspaceId: owner.id,
-          status: "active",
-        },
-      });
-
-      if (!secret) {
-        // Silently fail to avoid bruteforce.
-        res.status(204).end();
-        return;
-      }
-
-      await secret.update({
-        status: "disabled",
-      });
-
-      res.status(204).end();
-      return;
-
     case "POST":
       const { name: postSecretName } = req.body;
       const secretValue = req.body.value;
 
       const encryptedValue = encrypt(secretValue, owner.sId); // We feed the workspace sid as key that will be added to the salt.
 
-      await DustAppSecret.create({
-        userId: user.id,
-        workspaceId: owner.id,
-        name: postSecretName,
-        hash: encryptedValue,
-        status: "active",
+      let postSecret = await DustAppSecret.findOne({
```

Comment:
let's make this a lib/api/dust_app_secrets.ts method

Reply:
r ? https://github.com/dust-tt/dust/pull/4980 🙏